packages/wallet-connect-sdk-core/src/index.ts

```diff
@@ -30,63 +120,178 @@ export enum WitnessScope {
+ *   // the downside of this approach is that you will have to pass wcClient and session as parameter on most of the methods
+ * ```
+ */
 export class WcSdk {
+    /**
+     * Wallet Connect Client, used on most of the methods
+     */
     wcClient?: Client
+
+    /**
+     * The Session of the current connection
+     */
     session?: SessionTypes.Created
 
-    async initClient(logger: string, relayServer: string) {
+    /**
+     * Initializes the SDK, it's the first method to be called
+     * @param logger the logger level, describes how much information to show on the log, use `debug` for more information or `error` for less information
+     * @param relayServer the relayserver to connect to, it needs to be the same relay server of the wallet. It's recommended to use `wss://relay.walletconnect.org`
+     */
+    async initClient (logger: string, relayServer: string): Promise<void> {
         this.wcClient = await WcSdk.initClient(logger, relayServer)
     }
 
-    subscribeToEvents(callbacks?: WcCallbacks) {
+    /**
+     * Subscribe to Wallet Connect events
+     * ```
+     * wcInstance.subscribeToEvents({
+     *     onProposal: (uri: string) => {
+     *         // show the QRCode, you can use @walletconnect/qrcode-modal to do so, but any QRCode presentation is fine
+     *         QRCodeModal.open(uri, () => {})
+     *         // alternatively you can show Neon Wallet Connect's website, which is more welcoming
+     *         window.open(`https://neon.coz.io/connect?uri=${uri}`, '_blank').focus();
+     *     },
+     *     onDeleted: () => {
+     *         // here is where you describe a logout callback
+     *         logout()
+     *     }
+     * })
+     * ```
+     */
+    subscribeToEvents (callbacks?: WcCallbacks): void {
         WcSdk.subscribeToEvents(this.wcClient, callbacks)
     }
 
-    async loadSession() {
+    /**
+     * Load any existing connection, it should be called after the initialization, to reestablish connections made previously
+     */
+    async loadSession (): Promise<void> {
         if (this.wcClient) {
             this.session = await WcSdk.getSession(this.wcClient) || undefined
         }
     }
 
-    async connect(options: WcConnectOptions) {
+    /**
+     * Start the process of establishing a new connection, to be used when there is no `wcInstance.session`
+     * @param options describes the options for the connection
+     */
+    async connect (options: WcConnectOptions): Promise<void> {
         if (!this.wcClient) {
             throw Error('The client was not initialized')
         }
         this.session = await WcSdk.connect(this.wcClient, options)
     }
 
-    getAccountAddress(accountIndex?: number) {
+    /**
+     * gets the address of the connected account
+     * @param accountIndex the index of the account to retrieve, gets the first account if no index is provided
+     * @return the address of the connected account of the wallet

```

Comment:
Kind of confusing. Can we omit the word wallet here?
```suggestion
     * @return the address of the connected account.

```